temporal-sdk/src/main/java/io/temporal/internal/common/InternalUtils.java

```diff
@@ -93,10 +90,10 @@ public static Object getValueOrDefault(Object value, Class<?> valueClass) {
 
   public static SearchAttributes convertMapToSearchAttributes(
       Map<String, Object> searchAttributes) {
-    DataConverter converter = DataConverter.getDefaultInstance();
+    SearchAttributesPayloadConverter converter = new SearchAttributesPayloadConverter();
```

Comment:
No, there is no need to create a converter for each serialization. The creation of Jackson mappers is very expensive and losing caches inside it too. Please use the singleton pattern as we discussed.